autoload/OmniSharp/util.vim

```diff
@@ -275,7 +275,23 @@ function! OmniSharp#util#TranslatePathForClient(filename) abort
   if filename =~# '\$metadata\$'
     let filename = OmniSharp#util#TempDir() . '/' . fnamemodify(filename, ':t')
   endif
-  return fnamemodify(filename, ':.')
+
+  let modifiers = get(g:, 'OmniSharp_filename_modifiers', ':.')
+
+  if modifiers ==# 'relative'
+    let common = escape(getcwd(), '\')
+    let result = ''
+    while substitute(filename, common . s:dir_separator, '', '') ==# filename
+    \ && common !=# s:dir_separator
+      let common = fnamemodify(common, ':h')
+      let result .= '..' . s:dir_separator
+    endwhile
+    if common !=# s:dir_separator
```

Comment:
But this means that if there's no common subpath, we fall through, and then try "fnamemodify(filename, 'relative')`, which isn't right. Maybe else `let modifiers = ':p'`?

Reply:
You're right. It doesn't seem to cause errors this way but it's certainly not right, I'll make the change you've suggested.